coala_quickstart/green_mode/filename_operations.py

```diff
@@ -0,0 +1,141 @@
+import os
+from copy import deepcopy
+
+from coala_quickstart.green_mode.green_mode import (
+    append_to_contents)
+
+
+class Node:
+    def __init__(self, character, parent=None):
+        self.count = 1
+        self.character = character
+        self.children = {}
+        self.parent = parent
+
+    def insert(self, string, idx):
+        if idx >= len(string):
+            return
+        code = ord(string[idx])  # ASCII code
+        ch = string[idx]
+        if ch in self.children:
+            self.children[ch].count += 1
+        else:
+            self.children[ch] = Node(string[idx], self)
+        self.children[ch].insert(string, idx+1)
+
+
+class Trie:
+    """
+    Creates a Trie data structure for storing names of files.
+    """
+
+    def __init__(self):
+        self.root = Node('')
+
+    def insert(self, string):
+        self.root.insert(string, 0)
+
+    # Just a wrapper function.
+    def get_prefixes(self, min_length, min_files):
+        """
+        Discovers prefix from the Trie. Prefix shorter than the
+        min_length or matching against files lesser than the
+        min_files are not stored. Returns the prefixes in sorted
+        order.
+        """
+        self.prefixes = {}
+        self._discover_prefixes(self.root, [], min_length, 0, min_files)
+        return sorted(self.prefixes.items(), key=lambda x: (x[1], x[0]),
+                      reverse=True)
+
+    def _discover_prefixes(self, node, prefix, min_length, len, min_files):
+        """
+        Performs a DFA search on the trie. Discovers the prefixes in the trie
+        and stores them in the self.prefixes dictionary.
+        """
+        if node.count < min_files and node.character != '':
+            return
+        if len >= min_length:
+            current_prefix = ''.join(prefix)+node.character
+            to_delete = []
+            for i in self.prefixes:
+                if i in current_prefix:  # pragma: no cover
```

Comment:
why `# pragma: no cover` ? permanent or just temporary while you solved other problems ?